[Automation] native-image run fails for org.eclipse.jdt:ecj:3.29.0

## native-image run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: native-image run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=3.29.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.eclipse.jdt:ecj:3.29.0`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/26161670049/job/76955034320
Last 300 lines of the log:
```console
LANG=C.UTF-8 \
PATH=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/bin:/opt/hostedtoolcache/Java_GraalVM_jdk/21/x64/bin:/snap/bin:/home/runner/.local/bin:/opt/pipx_bin:/home/runner/.cargo/bin:/home/runner/.config/composer/vendor/bin:/usr/local/.ghcup/bin:/home/runner/.dotnet/tools:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin \
PWD=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0 \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/bin/java \
-XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI \
-Dgraalvm.locatorDisabled=true \
-Dtruffle.TruffleRuntime=com.oracle.truffle.api.impl.DefaultTruffleRuntime \
-Dgraalvm.ForcePolyglotInvalid=true \
-Dsubstratevm.HostLibC=glibc \
--add-exports=java.base/com.sun.crypto.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.access=jdk.graal.compiler.hostvmaccess,org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.event=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.foreign=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.loader=jdk.graal.compiler.hostvmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.logger=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.misc=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto \
--add-exports=java.base/jdk.internal.module=jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.shared \
--add-exports=java.base/jdk.internal.perf=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.platform=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.ref=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=java.base/jdk.internal.reflect=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.vm.annotation=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal.vm=org.graalvm.nativeimage.builder \
--add-exports=java.base/jdk.internal=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.invoke.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.net.www=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.net=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.nio.ch=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=java.base/sun.reflect.annotation=org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.factory=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.reflectiveObjects=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.repository=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.scope=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.reflect.generics.tree=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.jca=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.rsa=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.ssl=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.util=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.security.x509=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.text.spi=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.calendar=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.cldr=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale.provider=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.locale=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.resources=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util.spi=org.graalvm.nativeimage.builder \
--add-exports=java.base/sun.util=org.graalvm.nativeimage.builder \
--add-exports=java.management/com.sun.jmx.mbeanserver=org.graalvm.nativeimage.builder \
--add-exports=java.management/sun.management=org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.api.replacements=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.api.runtime=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.core.target=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.phases.util=jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.runtime=jdk.graal.compiler.hostvmaccess \
--add-exports=jdk.graal.compiler/jdk.graal.compiler.util.json=com.oracle.graal.reporter \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.aarch64=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.amd64=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.site=com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code.stack=jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.code=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.common=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.aarch64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.amd64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot.riscv64=jdk.graal.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.hotspot=com.oracle.graal.graal_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta.annotation=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.guest.staging,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.meta=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.configure,org.graalvm.nativeimage.guest.staging,org.graalvm.nativeimage.objectfile,org.graalvm.nativeimage.pointsto,org.graalvm.truffle.compiler \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.riscv64=com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.runtime=com.oracle.graal.graal_enterprise,jdk.graal.compiler,jdk.graal.compiler.hostvmaccess,org.graalvm.nativeimage.base,org.graalvm.nativeimage.builder,org.graalvm.nativeimage.pointsto \
--add-exports=jdk.internal.vm.ci/jdk.vm.ci.services=com.oracle.graal.graal_enterprise,com.oracle.svm.svm_enterprise,jdk.graal.compiler,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.events=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.event=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.jfc=com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.settings=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal.tracing=org.graalvm.nativeimage.builder \
--add-exports=jdk.jfr/jdk.jfr.internal=com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.builder \
--add-exports=jdk.management/com.sun.management.internal=org.graalvm.nativeimage.builder \
-XX:+UseJVMCINativeLibrary \
-XX:MaxRAMPercentage=84.99999999642142 \
'-Dsvm.build.memoryUsageReasonText=13.27GiB of memory (85% of system memory because $CI set to '"'"'true'"'"')' \
-Xss10m \
-XX:+UseParallelGC \
-XX:GCTimeRatio=9 \
-XX:+ExitOnOutOfMemoryError \
-XX:+UseCompressedOops \
-Djava.awt.headless=true \
'-Dorg.graalvm.vendor=Oracle Corporation' \
-Dorg.graalvm.vendorurl=https://www.graalvm.org/ \
'-Dorg.graalvm.vendorversion=Oracle GraalVM 25.1.0-dev+9.1' \
-Dorg.graalvm.version=25.1.0-dev \
-Dcom.oracle.graalvm.isaot=true \
-Djava.system.class.loader=com.oracle.svm.hosted.NativeImageSystemClassLoader \
-Djava.util.concurrent.ForkJoinPool.common.exceptionHandler=com.oracle.svm.hosted.CommonPoolUncaughtExceptionHandler \
'-Djava.util.concurrent.ForkJoinPool.common.threadFactory=com.oracle.svm.hosted.NativeImageSystemClassLoader$NativeImageForkJoinWorkerThreadFactory' \
-Dorg.graalvm.nativeimage.imagecode=buildtime \
-Xshare:off \
-Djdk.internal.lambda.disableEagerInitialization=true \
-Djdk.internal.lambda.eagerlyInitialize=false \
-Djava.lang.invoke.InnerClassLambdaMetafactory.initializeLambdas=false \
-Djava.lang.invoke.MethodHandle.DONT_INLINE_THRESHOLD=-1 \
-Djava.lang.invoke.MethodHandle.PROFILE_GWT=false \
-Djunit.platform.listeners.uid.tracking.output.dir=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/test-results/test/testlist \
-Dcom.oracle.svm.provided=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-shared.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest-staging.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/vmaccess-guest.jar \
--add-modules=ALL-DEFAULT \
--enable-native-access=org.graalvm.nativeimage.objectfile,org.graalvm.espresso.shared.svm,org.graalvm.nativeimage.builder,com.oracle.graal.reporter,org.graalvm.nativeimage.shared,jdk.graal.compiler.vmaccess,org.graalvm.nativeimage.guest.staging,jdk.graal.compiler.hostvmaccess,org.graalvm.nativeimage.configure,jdk.graal.compiler.vmaccess.guest,org.graalvm.nativeimage.foreign,org.graalvm.nativeimage.shaded.capnproto,com.oracle.svm.svm_enterprise,org.graalvm.nativeimage.pointsto,org.graalvm.nativeimage.base \
--module-path \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/reporter.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/pointsto.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm-shared.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/objectfile.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm-configure.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/hostvmaccess.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm-foreign.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/vmaccess.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/native-image-base.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm-enterprise.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm-capnproto-runtime.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/svm-guest-staging.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/espresso-svm.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/builder/vmaccess-guest.jar \
--module \
org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner \
-keepalive \
/proc/16221/comm \
-imagecp \
/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/libs/org.eclipse.jdt.ecj_tests.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.graalvm.buildtools/junit-platform-native/1.0.0/2974f834a2f0a0105f2284fba7ef16dc4d60de56/junit-platform-native-1.0.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter/5.9.2/26c586fbe0ebd81b48c9f11f0d998124248697ae/junit-jupiter-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-reporting/1.9.2/8bea292e2a546e4c21071f6bee60917ffd9f0101/junit-platform-reporting-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-launcher/1.9.2/38e0bfad8c57d4cd1a8f27926c25ffe9543068d6/junit-platform-launcher-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.vintage/junit-vintage-engine/5.9.2/53421816bde124a564a64ba005dcc0c8e66a9722/junit-vintage-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-params/5.9.2/bc2765afb7b85b583c710dd259a11c6b8c39e912/junit-jupiter-params-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-engine/5.9.2/572f7a553b53f83ee59cc045ce1c3772864ab76c/junit-jupiter-engine-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.jupiter/junit-jupiter-api/5.9.2/fed843581520eac594bc36bb4b0f55e7b947dda9/junit-jupiter-api-5.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-engine/1.9.2/40aeef2be7b04f96bb91e8b054affc28b7c7c935/junit-platform-engine-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-commons/1.9.2/6f9f8621d8230cd38aa42e58ccbc0c00569131ce/junit-platform-commons-1.9.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.junit.platform/junit-platform-console/1.9.2/5548148dd55e820c42bd817ab93c1eaa05ce37f6/junit-platform-console-1.9.2.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/tck-test-support/build/libs/tck-test-support-1.0.0-SNAPSHOT.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.eclipse.jdt/ecj/3.29.0/9cfd84117b075ee012e9cdc4df750876759504a7/ecj-3.29.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.assertj/assertj-core/3.22.0/c300c0c6a24559f35fa0bd3a5472dc1edcd0111e/assertj-core-3.22.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/junit/junit/4.13.2/8ac9e16d933b6fb43bc7f576336b8f4d7eb5ba12/junit-4.13.2.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.opentest4j/opentest4j/1.2.0/28c11eb91f9b6d8e200631d46e20a7f407f2a046/opentest4j-1.2.0.jar:/home/runner/.gradle/caches/modules-2/files-2.1/org.hamcrest/hamcrest-core/1.3/42a25dc3219429f0e5d060061f71acb49bf010a0/hamcrest-core-1.3.jar:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/classes/java/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/resources/test:/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/test-results/test/testlist \
-imagemp \
/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar:/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar \
-H:CLibraryPath=/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/clibraries/linux-amd64/glibc,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/clibraries/linux-amd64,/opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/clibraries \
-H:Path@driver=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/native/nativeTestCompile \
-H:BuildOutputJSONFile@user=/tmp/native-image-build-output.json \
-H:ReachabilityMetadataResources@file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/resources/test/META-INF/native-image/reachability-metadata.json+api=META-INF/native-image/reachability-metadata.json \
-H:FallbackThreshold@user+api=0 \
-H:Name@user+api=org.eclipse.jdt.ecj_tests-tests \
-H:ConfigurationFileDirectories@user=/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/native/generated/generateTestResourcesConfigFile,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.opentest4j/opentest4j/1.2.0,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-commons/1.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-params/5.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.jupiter/junit-jupiter-engine/5.8.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-engine/1.10.4,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-reporting/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.platform/junit-platform-launcher/1.9.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.junit.vintage/junit-vintage-engine/5.9.1,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/junit/junit/4.13.2,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.hamcrest/hamcrest-core/1.3,/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/metadata/org.eclipse.jdt/ecj/3.29.0 \
-H:Optimize@user+api=b \
-H:+ReportExceptionStackTraces@user \
-H:FutureDefaults@user+api=all \
-H:+AllowJRTFileSystem@user \
-H:Features@user+api=org.graalvm.junit.platform.JUnitPlatformFeature \
'-H:Class@explicit main-class=org.graalvm.junit.platform.NativeImageJUnitLauncher' \
-H:ImageBuildID@driver=5f6651d8-5c27-63ac-ff30-a030fb2d3c95 \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/thirdparty/native-image.properties+api=com.oracle.svm.thirdparty.gson.GsonFeature' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/library-support.jar!/META-INF/native-image/com.oracle.svm/polyglot/native-image.properties+api=com.oracle.svm.polyglot.groovy.GroovyIndyInterfaceFeature' \
'-H:AddExports@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=java.base/jdk.internal.misc=org.graalvm.nativeimage.guest' \
'-H:Features@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=com.oracle.svm.guest.features.jdk.hosted.JavaNetHttpFeature' \
'-H:ClassInitialization@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/native-image.properties+api=sun.net.httpserver.simpleserver:run_time' \
'-H:ReachabilityMetadataResources@jar:file:///opt/hostedtoolcache/graalvm-jdk-latest-ea_linux-x64_bin/25.0.0/x64/graalvm-25.1.0-dev+9.1/lib/svm/svm-guest.jar!/META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json+api=META-INF/native-image/org.graalvm.nativeimage.guest/reachability-metadata.json' \
-H:DriverWarningsCount=4
]
Warning: Option 'FallbackThreshold' is deprecated and might be removed in a future release: It no longer has any effect, and no replacement is available. Please refer to the GraalVM release notes.
========================================================================================================================
GraalVM Native Image: Generating 'org.eclipse.jdt.ecj_tests-tests' (executable)...
========================================================================================================================
For detailed information and explanations on the build output, visit:
https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/BuildOutput.md
------------------------------------------------------------------------------------------------------------------------
[1/8] Initializing...                                                                                   (7.0s @ 0.19GiB)
 Builder configuration:
 - Java version: 25.0.3+9-LTS, vendor version: Oracle GraalVM 25.1.0-dev+9.1
 - Graal compiler: optimization level: b, target machine: x86-64-v3, PGO: off
 - C compiler: gcc (linux, x86_64, 13.3.0)
 - Assertions: disabled, system assertions: disabled
 - 1 user-specific feature(s):
   - org.graalvm.junit.platform.JUnitPlatformFeature
 Image configuration:
 - Garbage collector: Serial GC (max heap size: 80% of RAM)
 - Assertions: disabled (class-specific config may apply), system assertions: disabled
------------------------------------------------------------------------------------------------------------------------
 2 experimental option(s) unlocked:
 - '-H:BuildOutputJSONFile' (origin(s): command line)
 - '-H:+AllowJRTFileSystem' (origin(s): command line)
------------------------------------------------------------------------------------------------------------------------
Build resources:
 - 13.27GiB of memory (85% of system memory because $CI set to 'true')
 - 4 thread(s) (100.0% of 4 available processor(s), determined at start)
[junit-platform-native] Running in 'test listener' mode using files matching pattern [junit-platform-unique-ids*] found in folder [/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/test-results/test/testlist] and its subfolders.
[2/8] Performing analysis...  [******]                                                                 (46.4s @ 1.39GiB)
   10,232 types,  15,204 fields, and  51,286 methods found reachable
      477 types,   1,176 fields, and   1,072 methods registered for reflection
       60 types,      68 fields, and      52 methods registered for JNI access
      253 resource accesses registered with 0.98MiB total size
        4 native libraries: dl, pthread, rt, z
[3/8] Building universe...                                                                              (6.7s @ 1.19GiB)
[4/8] Parsing methods...      [***]                                                                     (5.1s @ 1.29GiB)
[5/8] Inlining methods...     [***]                                                                     (2.7s @ 1.38GiB)
[6/8] Compiling methods...    [*****]                                                                  (31.2s @ 1.18GiB)
[7/8] Laying out methods...   [**]                                                                      (4.6s @ 1.49GiB)
[8/8] Creating image...       [**]                                                                      (4.1s @ 1.64GiB)
  17.62MiB (43.43%) for code area:    33,937 compilation units
  19.56MiB (48.21%) for image heap:  287,227 objects and 115 resources
   3.39MiB ( 8.36%) for other data
  40.58MiB in total image size, 37.75MiB in total file size
------------------------------------------------------------------------------------------------------------------------
Top 10 origins of code area:                                Top 10 object types in image heap:
   1.83MiB org.eclipse.jdt.internal.compiler.lookup            5.06MiB byte[] for code metadata
   1.77MiB org.eclipse.jdt.internal.compiler.ast               3.29MiB byte[] for string data
   1.40MiB o.g.n.runtime/com.oracle.svm.core.code              1.74MiB java.base/java.lang.String
 833.58KiB org.eclipse.jdt.internal.compiler.parser            1.33MiB java.base/java.lang.Class
 749.83KiB java.base/java.util                                 1.09MiB o.g.n.runtime/c.o.s.core.hub.DynamicHubCompanion
 447.05KiB org.eclipse.jdt.internal.compiler.problem           0.99MiB byte[] for embedded resources
 430.98KiB org.eclipse.jdt.internal.compiler.codegen         984.31KiB byte[] for general heap data
 414.73KiB java.base/jdk.internal.classfile.impl             491.36KiB byte[] for reflection metadata
 403.65KiB org.eclipse.jdt.internal.compiler.batch           467.43KiB heap alignment
 359.63KiB org.eclipse.jdt.internal.compiler                 397.19KiB o.g.n.runtime/c.o.s.c.h.ImageReflectionMetadata
   9.06MiB for 349 more packages                               3.78MiB for 2408 more object types
                            Use '--emit build-report' to create a report with more details.
------------------------------------------------------------------------------------------------------------------------
Security report:
 - Binary includes Java deserialization.
 - CycloneDX SBOM with 19 component(s) is embedded in binary (1.07KiB). 35 type(s) could not be associated to a component.
 - Advanced obfuscation not enabled; enable with '-H:AdvancedObfuscation=""' (experimental support).
------------------------------------------------------------------------------------------------------------------------
Recommendations:
 G1GC: Use the G1 GC ('--gc=G1') for improved latency and throughput.
 PGO:  Use Profile-Guided Optimizations ('--pgo') for improved throughput.
 HEAP: Set max heap for improved and more predictable memory usage.
 CPU:  Enable more CPU features with '-march=native' for improved performance.
------------------------------------------------------------------------------------------------------------------------
                       8.9s (8.0% of total time) in 1261 GCs | Peak RSS: 2.33GiB | CPU load: 3.66
------------------------------------------------------------------------------------------------------------------------
Build artifacts:
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/native/nativeTestCompile/libmanagement_ext.so (jdk_library, 32.23KiB)
 /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/native/nativeTestCompile/org.eclipse.jdt.ecj_tests-tests (executable, 37.75MiB)
 /tmp/native-image-build-output.json (build_info, 1.03KiB)
========================================================================================================================
Finished generating 'org.eclipse.jdt.ecj_tests-tests' in 1m 50s.

The build process encountered 5 warnings.

[native-image-plugin] Native Image written to: /home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/native/nativeTestCompile
Exception in thread "main" java.lang.ExceptionInInitializerError
	at java.base@25.0.3/jdk.internal.module.SystemModuleFinders$SystemImage.reader(SystemModuleFinders.java:129)
	at java.base@25.0.3/jdk.internal.module.SystemModuleFinders$SystemModuleReader.containsImageLocation(SystemModuleFinders.java:410)
	at java.base@25.0.3/jdk.internal.module.SystemModuleFinders$SystemModuleReader.find(SystemModuleFinders.java:93)
	at java.base@25.0.3/jdk.internal.loader.BuiltinClassLoader.findMiscResource(BuiltinClassLoader.java:447)
	at java.base@25.0.3/jdk.internal.loader.BuiltinClassLoader.findResources(BuiltinClassLoader.java:377)
	at java.base@25.0.3/jdk.internal.loader.BootLoader.findResources(BootLoader.java:178)
	at java.base@25.0.3/java.lang.ClassLoader.getResources(ClassLoader.java:1410)
	at java.base@25.0.3/java.lang.ClassLoader.getResources(ClassLoader.java:1408)
	at java.base@25.0.3/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1074)
	at java.base@25.0.3/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1099)
	at java.base@25.0.3/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1142)
	at java.base@25.0.3/java.util.ServiceLoader$1.hasNext(ServiceLoader.java:1164)
	at java.base@25.0.3/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1246)
	at java.base@25.0.3/java.lang.Iterable.forEach(Iterable.java:74)
	at org.junit.platform.launcher.core.LauncherFactory.collectTestEngines(LauncherFactory.java:145)
	at org.junit.platform.launcher.core.LauncherFactory.createDefaultLauncher(LauncherFactory.java:131)
	at org.junit.platform.launcher.core.LauncherFactory.create(LauncherFactory.java:125)
	at org.junit.platform.launcher.core.LauncherFactory.create(LauncherFactory.java:109)
	at org.graalvm.junit.platform.NativeImageJUnitLauncher.main(NativeImageJUnitLauncher.java:118)
	at java.base@25.0.3/java.lang.invoke.LambdaForm$DMH/sa346b79c.invokeStaticInit(LambdaForm$DMH)
Caused by: java.lang.NullPointerException
	at java.base@25.0.3/java.util.Objects.requireNonNull(Objects.java:220)
	at java.base@25.0.3/sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:257)
	at java.base@25.0.3/jdk.internal.jimage.ImageReaderFactory.<clinit>(ImageReaderFactory.java:66)
	... 20 more

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeTest'.
> Process 'command '/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/native/nativeTestCompile/org.eclipse.jdt.ecj_tests-tests'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

BUILD FAILED in 1m 56s

> Task :nativeTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.eclipse.jdt/ecj/3.26.0/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
18 actionable tasks: 8 executed, 10 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for org.eclipse.jdt:ecj:3.29.0 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1m 57s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 3.29.0 at 2026-05-20T13:10:15Z with exit code 1 after 118s
❌ ERROR: 'native-image run' for 3.29.0 failed with exit code 1.
FAILED[native-image run][3.29.0][env GVM_TCK_LV=3.29.0 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean test -Pcoordinates=org.eclipse.jdt:ecj:3.29.0]
```

--- a/metadata/org.eclipse.jdt/ecj/3.29.0/reachability-metadata.json
+++ b/metadata/org.eclipse.jdt/ecj/3.29.0/reachability-metadata.json
@@ -4174,6 +4174,12 @@
     },
     {
       "condition" : {
+        "typeReached" : "org.eclipse.jdt.internal.compiler.tool.JrtFileSystem"
+      },
+      "glob" : "META-INF/services/java.nio.file.spi.FileSystemProvider"
+    },
+    {
+      "condition" : {
         "typeReached" : "org.eclipse.jdt.internal.compiler.batch.Main$ResourceBundleFactory"
       },
       "glob" : "META-INF/services/java.util.spi.ResourceBundleControlProvider"

--- a/stats/org.eclipse.jdt/ecj/3.29.0/stats.json
+++ b/stats/org.eclipse.jdt/ecj/3.29.0/stats.json
@@ -20,15 +20,15 @@
     },
     "libraryCoverage" : {
       "instruction" : {
-        "covered" : 55373,
-        "missed" : 478134,
-        "ratio" : 0.103791,
+        "covered" : 55332,
+        "missed" : 478175,
+        "ratio" : 0.103714,
         "total" : 533507
       },
       "line" : {
-        "covered" : 12778,
-        "missed" : 108253,
-        "ratio" : 0.105576,
+        "covered" : 12770,
+        "missed" : 108261,
+        "ratio" : 0.10551,
         "total" : 121031
       },
       "method" : {

